_includes/_components/user.html

```diff
@@ -1,51 +1,150 @@
-<!-- TODO -->
-
-      <article class="component dark-blue">
-        <header>
-          <div class="container">
+<article class="component dark-blue">
+    <header>
+        <div class="container">
             <h2>Users</h2>
-            <p>This component blablabla</p>
-          </div>
-        </header>
-        <div class="container content">
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
-          </p>
-          <div class="row">
+            <p>A simple way to handle password encryption and salt generation.</p>
+        </div>
+    </header>
+    <div class="container content">
+        <div class="row">
             <div class="col-md-6">
-              <div class="standard">
-                <div class="banner"></div>
+                <div class="standard">
+                    <div class="banner"></div>
 {% highlight php %}
 <?php
 
-class Test
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class User implements UserInterface
 {
-    public function test(SomeClass $subject)
+    private $id;
+
+    private $password;
+
+    private $salt;
+
+    // ...
+
+    public function setPassword($password)
+    {
+        $this->password = $password;
+
+        return $this->password;
+    }
+
+    public function setSalt($salt)
```

Comment:
See [here](http://symfony.com/doc/current/cookbook/security/entity_provider.html).

You generally don't add a setSalt() method. You either generate a unique salt in constructor or you just don't (if you use bcrypt).